packages/taiko-client/prover/proof_producer/sgx_producer.go

```diff
@@ -113,47 +112,43 @@ func (s *SGXProofProducer) callProverDaemon(ctx context.Context, opts *ProofRequ
 		proof []byte
 		start = time.Now()
 	)
-	if err := backoff.Retry(func() error {
-		if ctx.Err() != nil {
-			return nil
-		}
-		output, err := s.requestProof(opts)
-		if err != nil {
-			log.Error("Failed to request proof", "height", opts.BlockID, "error", err, "endpoint", s.RaikoHostEndpoint)
-			return err
-		}
-
-		if output == nil {
-			log.Info(
-				"Proof generating",
-				"height", opts.BlockID,
-				"time", time.Since(start),
-				"producer", "SGXProofProducer",
-			)
-			return errProofGenerating
-		}
-
-		log.Debug("Proof generation output", "output", output)
-
-		// Raiko returns "" as proof when proof type is native,
-		// so we just convert "" to bytes
-		if s.ProofType == ProofTypeCPU {
-			proof = common.Hex2Bytes(output.Data.Proof)
-		} else {
-			proof = common.Hex2Bytes(output.Data.Proof[2:])
-		}
 
+	if ctx.Err() != nil {
+		return nil, ctx.Err()
+	}
+	output, err := s.requestProof(opts)
+	if err != nil {
+		log.Error("Failed to request proof", "height", opts.BlockID, "error", err, "endpoint", s.RaikoHostEndpoint)
+		return nil, err
+	}
+
+	if output == nil {
 		log.Info(
-			"Proof generated",
+			"Proof generating",
 			"height", opts.BlockID,
 			"time", time.Since(start),
 			"producer", "SGXProofProducer",
 		)
-		return nil
-	}, backoff.WithContext(backoff.NewConstantBackOff(proofPollingInterval), ctx)); err != nil {
-		return nil, err
+		return nil, errProofGenerating
```

Comment:
looks like `errProofGenerating` is never used now?

Reply:
Yes, after raiko remove the native type, it is never used.

Reply:
And I will refactor the code in other PR.